Objective-C/Tests/DatabaseTest.m

```diff
@@ -582,6 +582,59 @@ - (void) testSaveDocWithDeletedConflictUsingConcurrencyControl: (int)concurrency
     [self cleanDB];
 }
 
+#pragma mark Save Conflict Handler
+
+- (void) testConflictHandler {
+    NSString* docID = @"doc1";
+    CBLMutableDocument* doc = [[CBLMutableDocument alloc] initWithID: docID];
+    [doc setString: @"Tiger" forKey: @"firstName"];
+    [self saveDocument: doc];
+    AssertEqual([self.db documentWithID: docID].generation, 1u);
+    
+    CBLMutableDocument* doc1a = [[self.db documentWithID: docID] toMutable];
+    CBLMutableDocument* doc1b = [[self.db documentWithID: docID] toMutable];
+    
+    [doc1a setString: @"Scotty" forKey: @"nickName"];
+    [self saveDocument: doc1a];
+    AssertEqual([self.db documentWithID: docID].generation, 2u);
+    
+    NSError* error;
+    [doc1b setString: @"Scott" forKey: @"nickName"];
+    [self.db saveDocument: doc1b
+          conflictHandler:^BOOL(CBLMutableDocument * cur, CBLDocument * old) {
+              AssertEqualObjects(doc1b.toDictionary, cur.toDictionary);
+              AssertEqualObjects(doc1a.toDictionary, old.toDictionary);
+              AssertEqual(cur.generation, 2u);
+              AssertEqual(old.generation, 2u);
+              return YES;
+          } error: &error];
+    AssertNil(error);
```

Comment:
Again, test the return value of `saveDocument`, don't test `error`.

Reply:
intention here is to check return value of the `saveDocument` and error = nil. So that no unnecessary error is returned in case of success. 

```
NSError* error = nil;
....
Assert([save document: &error]);
AssertNil(error);
```